src/lib/core/datetime.c

```diff
@@ -197,7 +199,13 @@ datetime_to_string(const struct datetime *date, char *buf, ssize_t len)
 			SNPRINT(sz, snprintf, buf, len, ".%09d", nanosec);
 		}
 	}
-	if (offset == 0) {
+	if (tzindex != 0) {
+		const char *tz_name = timezone_name(tzindex);
+		assert(tz_name != NULL);
+		SNPRINT(sz, snprintf, buf, len,
+			tz_name[1] == '\0' ? "%s" : " %s",
+			tz_name);
+	} else if (offset == 0) {
 		SNPRINT(sz, snprintf, buf, len, "Z");
```

Comment:
Yup, that's default mode, showing UTC time without any requested (human-readable) timezone name. i.e. when one entered
`2020-04-22T00:00` it will be interpreted as UTC and will be displayed as Zulu(UTC) intentionally, to highlight implicit offset.

The difference with our case - when customer intentionally entered `2020-04-22T00:00UTC` we should keep that `UTC` and display as-is (same applies to `GMT`, or `Z` zones). Despite the fact they map to the same offset, we will try to retain original spelling.

Reply:
Ok, I see. Thanks!